sharding/sharding.go

```diff
@@ -286,3 +300,60 @@ func compileRegexps(exps []string) ([]*regexp.Regexp, error) {
 
 	return res, nil
 }
+
+func (r *ShardingFerry) sanityCheckReplicationConfig() error {
```

Comment:
Apart from checking if the query is non-empty, we can definitely sanity check this inside ghostferry/ferry instead of in Sharding only. We can ship this as is and I'll refactor it when I bring this support to copydb soon.